DigitalExcellence/dex-backend
# Feature#438 transfer project ownership

## Description

Remember, this is a draft PR.
You will need two accounts to test this PR. 1 account to initiate the project transfer from and the second one to take the ownership. You also will need 2 email accounts, So you could use your FHICT email and a personal mail etc.
You can change an email adress of a DeX user in the User table.

Also you will need to set envoirement variables for the sendgrid api key and the dex email account. You will find the name of the variables in the MailClient.cs class and the values you can ask me for or view our pass manager.
Make sure that your two email accounts are present in the database and that it does not show <Redacted> !!

## Type of change


## Checklist


## Steps to Test or Reproduce


1. Login as a user and check that you have a project.
2. Via SwaggerUI go to the **/api/Project/transfer/{projectId}** endpoint and enter the email address for the new project owner combined with the projectId. The email adress should belong to a DeX user. You can change the email address in the database. Or more easy use two different FHICT accounts.
3. You should have an email indicating the project transfer. Accept or Deny. Please test all possible outcomes.
4. After accepting the transfer, the new user should also receive an email (this is the email address you entered, this also must be a DeX user!!)
5. After accepting the second email verification the project should now be transferred to the new user.

## Link to issue

Closes: #438
